policyengine_us/variables/gov/states/ca/tax/income/credits/foster_youth/ca_fytc.py

```diff
@@ -0,0 +1,23 @@
+from policyengine_us.model_api import *
+
+
+class ca_fytc(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "California Foster Youth Tax Credit"
+    unit = USD
+    definition_period = YEAR
+    defined_for = StateCode.CA
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(period).gov.states.ca.tax.income.credits.foster_youth
+
+        max_amount = add(tax_unit, period, ["ca_fytc_eligible"])
+
+        earned_income = add(tax_unit, period, ["earned_income"])
+
+        reduction_amount = max_(
+            0, (earned_income - p.income_threshold) * p.marginal_rate
+        )
```

Comment:
```suggestion
        excess_earned_income = earned_income - p.income_threshold
        reduction_amount = max_(
            0, excess_earned_income * p.marginal_rate
        )
```

Reply:
I added a few lines to check whether the age_eligible is 0 or 25000 but I'm not sure if we need that. @PavelMakarchuk

Reply:
> I added a few lines to check whether the age_eligible is 0 or 25000 but I'm not sure if we need that. @PavelMakarchuk

we do need unit tests that cover every edge case, but I think the tests are failing